### ISSUE TITLE
Deprecation warning on lr_scheduler.step(num_steps)

### ISSUE THREAD

**Issue by MicaelCarvalho** [pytorch/pytorch#103444]
### 🐛 Describe the bug

`step(num_steps)` produces a deprecation warning currently. However, there is a legitimate use case for this API in learning rate schedulers — if reloading a trained model and continuing to train, it is necessary to advance the number of steps inside the scheduler to match the current model state.

The scheduler does not provide an alternative way to advance the number of steps. And it is not possible to use `state_dict()`  + `load_state_dict()` because they also prevent changing the learning rate and other hyperparameters during transitions, unless the user manually changes the state dict, which is a hacky.

```
UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
```

### Versions

Versions of relevant libraries:
[pip3] torch==2.0.1

**Comment by vadimkantorov** [comment]
Related: https://github.com/pytorch/pytorch/issues/68332 (if we have functional forms, more of these problems with state-passing can at least be worked around in a cleaner way)

**Comment by janeyx99** [comment]
After a brief git blame, the PR that introduced this warning is https://github.com/pytorch/pytorch/pull/31125, which was written 4 years ago in an attempt to fix another issue with silent incorrectness. 

We currently do not have a dedicated maintainer for LRScheduler, but as you may have guessed, this feature is likely not actually getting deprecated any time soon. One actionable thing to do is to change the DeprecationWarning to just a normal UserWarning, since it does warn against certain behavior that may lead to confusion.

I'll mark this issue as actionable for any community member to take up in the meantime.

**Comment by MicaelCarvalho** [comment]
Another case:
```
Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
```

We need to call `lr_scheduler.step(num_steps)` before training so we can advance the lr_scheduler back to its original position. Can't load the state dict because we may want to change other things in the schedule

**Comment by janeyx99** [comment]
We do not have any plans soon to implement the deprecation. However, for my understanding, another alternative to calling `lr_scheduler.step(num_steps)` is to just modify the state directly, yes?
